tests/providers/test_openai.py

```diff
@@ -0,0 +1,63 @@
+import os
+
+import pytest
+
+from exchange import Text, ToolUse
+from exchange.providers.openai import OpenAiProvider
+from .conftest import complete, vision, tools
+
+OPENAI_MODEL = os.getenv("OPENAI_MODEL", "gpt-4o-mini")
+
+
+@pytest.mark.vcr()
+def test_openai_complete(default_openai_env):
+    reply_message, reply_usage = complete(OpenAiProvider, OPENAI_MODEL)
+
+    assert reply_message.content == [Text(text="Hello! How can I assist you today?")]
+    assert reply_usage.total_tokens == 27
+
+
+@pytest.mark.integration
+def test_openai_complete_integration():
+    reply = complete(OpenAiProvider, OPENAI_MODEL)
+
+    assert reply[0].content is not None
+    print("Completion content from OpenAI:", reply[0].content)
+
+
+@pytest.mark.vcr()
```

Comment:
while these tests down look redundant to the ones in the parent dir, what's nice is the non_integration ones are using recorded requests. so we can see exactly what should be happening. This has been helpful figuring out what should work in azure openai, because it works in normal openai, but doesn't